pytorch_lightning/accelerators/tpu.py

```diff
@@ -55,3 +56,11 @@ def all_gather(self, tensor: torch.Tensor, group: Optional[Any] = None, sync_gra
         if torch.distributed.is_initialized():
             return xm.all_gather(tensor, group=group, sync_grads=sync_grads)
         return tensor
+
+    def clip_gradients(self, optimizer: Optimizer, grad_clip_val: Union[float, int], norm_type: float = 2.0):
```

Comment:
The patched XLA function does not do a check for `grad_clip_val>0` so we might need to include this as done here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/plugins/precision/precision_plugin.py#L109-L110

In addition, will this mean we can get rid of the TODO here? https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/plugins/precision/precision_plugin.py#L103

The implementation of clip gradients is TPU compliant I think, but I think that's not an issue.

Reply:
Good point !

Reply:
@SeanNaren The issue for `aitextgen`?

The code hangs when it calls xm.save in Transformer’s [save_pretrained](https://github.com/minimaxir/aitextgen/blob/master/aitextgen/train.py#L235) method, when passed xm_save to the save_function param from aitextgen for TPUs.
Code: https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_utils.py#L835

`xm.save` has [rendezvous](https://github.com/pytorch/xla/blob/master/torch_xla/core/xla_model.py#L819) in it. All the core were not syncing up for rendezvous. Hence, the code was hanging.